lib/model.js

```diff
@@ -3573,7 +3617,19 @@ class Model {
 
         this.setDataValue(field, values[field]);
 
-        return this.sequelize.getQueryInterface().update(this, this.constructor.getTableName(options), values, where, _.defaults({ hooks: false }, options));
+        return this.sequelize.getQueryInterface().update(
+          this, this.constructor.getTableName(options), values, where, _.defaults({ hooks: false }, options)
+        ).then(results => {
+          const rowsUpdated = results[1];
+          if (this.constructor._versionAttribute && rowsUpdated < 1) {
+            throw new sequelizeErrors.OptimisticLockError({
```

Comment:
Did not pass the model directly because it doesn't serialize as cleanly.  The reasoning is to pass only the necessary information needed to debug any potential errors.